applications/FluidDynamicsApplication/python_scripts/rotating_frame_process.py

```diff
@@ -0,0 +1,131 @@
+import KratosMultiphysics as KM
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+import numpy as np
+
+def Factory(settings, model):
+    if not isinstance(settings, KM.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return RotatingFrameProcess(model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class RotatingFrameProcess(KM.Process):
+    """A Kratos process that rotates a given RotatingFrameModelPart 
+    around a specified axis of rotation. This process uses the Kratos 
+    RotateNodesUtility to rotate the nodes of the RotatingFrameModelPart 
+    and assigns the corresponding MeshDisplacement to the rotated nodes. 
+    The process also assigns the corresponding rotational mesh velocity to 
+    the nodes in a RotatingObjectModelPart. The process takes as input the 
+    model part to be rotated, the axis of rotation vector, the final angular 
+    velocity in radians per second, and the accelerating time in seconds.
+
+    Public member variables:
+    model -- the container of the different model parts.
+    settings -- Kratos parameters containing process settings.
+    """
+
+    def __init__(self, model, settings):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing process settings.
+        """
+        KM.Process.__init__(self) # calling the baseclass constructor
+
+        default_settings = KM.Parameters("""{
+            "rotating_frame_model_part_name": "",
+            "rotating_object_model_part_name": "",
```

Comment:
Which is the difference between these two? Aren't you doing the rotation in the entire model part?

Reply:
The `rotating_frame_model_part_name` refers to the computational domain treated as being in a rotational state, typically employed in the sliding mesh approach to capture the effects of the rotating zone. On the other hand, the `rotating_object_model_part_name` denotes the specific entity or component, like a turbine blade or a rotor, that necessitates the assignment of the rotational velocity. Essentially, while the rotating frame captures the overall moving region in the computational sense, the rotating object represents the actual physical component causing this rotation.